src/Modules/Settings/Dnn.PersonaBar.TaskScheduler/Services/TaskSchedulerController.cs

```diff
@@ -362,12 +362,13 @@ public HttpResponseMessage UpdateScheduleItem(ScheduleDto scheduleDto)
                 updatedItem.ScheduleID = scheduleDto.ScheduleID;
 
 
-                if ((updatedItem.ScheduleStartDate != existingItem.ScheduleStartDate) ||
-                    (updatedItem.Enabled) ||
-                    (updatedItem.TimeLapse != existingItem.TimeLapse) ||
-                    (updatedItem.RetryTimeLapse != existingItem.RetryTimeLapse) ||
-                    (updatedItem.RetryTimeLapseMeasurement != existingItem.RetryTimeLapseMeasurement) ||
-                    (updatedItem.TimeLapseMeasurement != existingItem.TimeLapseMeasurement))
+                if (updatedItem.ScheduleStartDate != existingItem.ScheduleStartDate ||
+                    updatedItem.Enabled ||
+                    updatedItem.Enabled != existingItem.Enabled ||
+                    updatedItem.TimeLapse != existingItem.TimeLapse ||
+                    updatedItem.RetryTimeLapse != existingItem.RetryTimeLapse ||
+                    updatedItem.RetryTimeLapseMeasurement != existingItem.RetryTimeLapseMeasurement ||
+                    updatedItem.TimeLapseMeasurement != existingItem.TimeLapseMeasurement)
```

Comment:
By the way, my head spins from all those statements. Maybe the condition can be simplified?

Reply:
hmm, looks like need to check whether these properties has been changed one by one.

Reply:
You say that we may be missing doing the wrong update in some cases?

Reply:
I can't find out a better way to check all these conditions. :(